### PR TITLE
Add Support for HRI Features

### PR DESCRIPTION
**Pull request by developerkunal**
### 🔧 Changes
- **Support for HRI Features Added**:
  - **Client Struct Enhancements**:
    - **SignedRequestObject**:
      - **Required**: Indicates if JWT-secured Authorization Requests (JAR) are mandatory.
      - **Credentials**: Specifies credentials for signing JAR requests.
    - **ComplianceLevel**: Sets the compliance level for the client, affecting its capabilities. To remove, use a PATCH request.
    - **RequireProofOfPossession**: Makes Proof-of-Possession mandatory (default: false).
  - **Credential Struct Additions**:
    - **SubjectDN**: Subject Distinguished Name, mutually exclusive with `pem` property.
    - **ThumbprintSHA256**: SHA256 thumbprint of the x509_cert certificate.
  - **ClientAuthenticationMethods Struct**:
    - **TLSClientAuth**:
      - **Credentials**: List of unique credential IDs for CA-based mTLS.
    - **SelfSignedTLSClientAuth**:
      - **Credentials**: List of unique credential IDs for self-signed mTLS.
  - **Resource Struct Enhancements**:
    - **ConsentPolicy**: Specifies the consent policy for the resource server. Options include:
      - `"transactional-authorization-with-mfa"`
      - `null` (to unset)
    - **AuthorizationDetails**:
      - **Type**: Identifier for the authorization detail type.
    - **TokenEncryption**: Specifies token encryption settings. Options include:
      - `"compact-nested-jwe"`
      - `null` (to unset)
      - **EncryptionKey**:
        - **Name**: Name of the encryption key.
        - **Alg**: Encryption algorithm.
        - **Kid**: Key ID.
        - **Pem**: PEM-formatted public key (JSON escaped).
    - **ProofOfPossession**:
      - **Mechanism**: Intended Proof-of-Possession mechanism, such as `"mtls"`.
      - **Required**: Whether Proof-of-Possession is mandatory.
  - **Tenant Struct Enhancements**:
    - **AcrValuesSupported**: List of supported ACR values. 
    - **MTLS**:
      - **EnableEndpointAliases**: If `true`, enables mTLS endpoint aliases.
    - **PushedAuthorizationRequestsSupported**: Enables Pushed Authorization Requests.
  - **TenantFlags Enhancements**:
    - **RemoveAlgFromJWKS**: Removes the `alg` property from the JWKS `.well-known` endpoint.

### 📚 References

- Auth0 Docs:
  - [Client Settings API](https://auth0.com/docs/api/management/v2/clients/get-clients)
  - [Tenant Settings API](https://auth0.com/docs/api/management/v2/tenants/tenant-settings-route)
  - [Resource Server Settings API](https://auth0.com/docs/api/management/v2/resource-servers/get-resource-servers)
 

### 🔬 Testing

- **Client Settings**: Verified HRI features are applied and functional in client settings.
- **Tenant Settings**: Confirmed HRI settings are properly configured and operational in tenant settings.
- **Resource Servers**: Ensured HRI features are integrated and functioning with resource server settings.

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] Documentation for all new/changed functionality is included (or N/A)

### CODE REVIEW

**Review by gyaneshgouraw-okta — APPROVED** [review]

**Comment by gyaneshgouraw-okta** [comment]
Lets add a bit more detail to PR description. We should add 

- Add detail/link which has some description about HRI
- We can document management api corresponding docs link
- Any other relevant doc which can assist users

File: management/tenant_test.go [hunk 1]
```diff
@@ -41,6 +41,11 @@ func TestTenantManager(t *testing.T) {
 		Sessions: &TenantSessions{
 			OIDCLogoutPromptEnabled: auth0.Bool(false),
 		},
+		AcrValuesSupported:                   &[]string{"foo", "bar"},
```
Comment (acwest) [inline management/tenant_test.go:44]: we need tests for both setting this to null in the api, and omitting it in the API

**Review by acwest — CHANGES_REQUESTED** [review]
more testing of new values, particularly `AcrValuesSupported`

**Review by acwest — APPROVED** [review]
